evm/client/compatibility_helper.go

```diff
@@ -55,3 +56,35 @@ func ToBackwardCompatibleCallArg(msg ethereum.CallMsg) interface{} {
 	}
 	return arg
 }
+
+// TODO: Cleanup, and rename this function
+func toBackwardCompatibleCallArgWithTronSupport(msg ethereum.CallMsg, chainType chaintype.ChainType) interface{} {
+	arg := map[string]interface{}{
+		"from": msg.From,
+		"to":   msg.To,
+	}
+	if len(msg.Data) > 0 {
+		if chainType == chaintype.ChainTron {
+			arg["data"] = hexutil.Bytes(msg.Data)
+		} else {
+			arg["input"] = hexutil.Bytes(msg.Data)
+			arg["data"] = hexutil.Bytes(msg.Data) // duplicate legacy field for compatibility
+		}
```

Comment:
Tron doesn't work with `"input"` in the message?

Let's compose the underlying existing `ToBackwardCompatibleCallArg` - call that one first, then, in case of Tron, remove `arg["input"]` from the `arg` output.